Shared/Samples/Run valve isolation trace/RunValveIsolationTraceView.Model.swift

```diff
@@ -0,0 +1,382 @@
+                        tracingActivity = .none
+                        
+                        // Get available utility categories.
+                        if let definition = utilityNetwork.definition {
+                            filterBarrierCategories = definition.categories
+                        }
+                    }
+                }
+            }
+        }
+        
+        /// When the utility network is loaded, create a`UtilityElement`
+        /// from the asset type to use as the starting location for the trace.
+        private func makeStartingLocation() -> UtilityElement? {
+            // Constants for creating the default starting location.
+            let networkSourceName = "Gas Device"
+            let assetGroupName = "Meter"
+            let assetTypeName = "Customer"
+            let terminalName = "Load"
+            let globalID = UUID(uuidString: "98A06E95-70BE-43E7-91B7-E34C9D3CB9FF")!
+            
+            // Create a default starting location.
+            if let networkSource = utilityNetwork.definition?.networkSource(named: networkSourceName),
+               let assetType = networkSource.assetGroup(named: assetGroupName)?.assetType(named: assetTypeName),
+               let startingLocation = utilityNetwork.makeElement(assetType: assetType, globalID: globalID) {
+                // Set the terminal for the location. (For our case, use the "Load" terminal.)
+                startingLocation.terminal = assetType.terminalConfiguration?.terminals.first(where: { $0.name == terminalName })
+                return startingLocation
+            } else {
+                return nil
+            }
+        }
+        
+        /// Adds a graphic to the graphics overlay.
+        /// - Parameters:
+        ///   - location: The `Point` location to place the graphic..
+        ///   - traceLocationType: The textual description of the trace location type.
+        private func addGraphic(for location: Point, traceLocationType: String) {
+            parametersOverlay.addGraphic(
+                Graphic(
+                    geometry: location,
+                    attributes: ["TraceLocationType": traceLocationType],
+                    symbol: nil
+                )
+            )
+        }
+        
+        /// Sets the selected filter barrier category and updates the status text.
+        func selectCategory(_ category: UtilityCategory) {
+            selectedCategory = category
+            traceEnabled = true
+            statusText = "\(category.name) selected."
+        }
+        
+        /// Runs a trace with the pending trace configuration and selects features in the map that
+        /// correspond to the element results.
+        @MainActor
+        func trace() async throws {
+            tracingActivity = .runningTrace
+            
+            guard let configuration = makeTraceConfiguration(category: selectedCategory) else {
```

Comment:
early return here - should `tracingActivity` be reset?  How about `resetEnabled`?